src/test/java/uk/gov/hmcts/reform/unspec/service/CoreCaseUserServiceTest.java

```diff
@@ -0,0 +1,104 @@
+import uk.gov.hmcts.reform.unspec.config.CrossAccessUserConfiguration;
+import uk.gov.hmcts.reform.unspec.enums.CaseRole;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.clearInvocations;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = {
+    CoreCaseUserService.class
+})
+class CoreCaseUserServiceTest {
+
+    private static final String USER_AUTH_TOKEN = "Bearer user-xyz";
+    private static final String DEFENDANT_USER_AUTH_TOKEN = "Bearer defendant-xyz";
+    private static final String SERVICE_AUTH_TOKEN = "Bearer service-xyz";
+    private static final String CASE_ID = "1";
+    private static final String USER_ID = "User1";
+    public static final String ORG_ID = "62LYJRF";
+
+    @MockBean
+    private CrossAccessUserConfiguration userConfig;
+
+    @MockBean
+    private CaseAccessDataStoreApi caseAccessDataStoreApi;
+
+    @MockBean
+    private IdamClient idamClient;
+
+    @MockBean
+    private OrganisationService organisationService;
+
+    @MockBean
+    private AuthTokenGenerator authTokenGenerator;
+
+    @Mock
+    private Organisation organisation;
+
+    @Autowired
+    private CoreCaseUserService service;
+
+    @BeforeEach
+    void init() {
+        clearInvocations(authTokenGenerator);
+        clearInvocations(idamClient);
+        when(authTokenGenerator.generate()).thenReturn(SERVICE_AUTH_TOKEN);
+        when(idamClient.getAccessToken(userConfig.getUserName(), userConfig.getPassword())).thenReturn(USER_AUTH_TOKEN);
+        when(idamClient.getUserInfo(DEFENDANT_USER_AUTH_TOKEN)).thenReturn(UserInfo.builder().uid(USER_ID).build());
+        when(organisationService.findOrganisation(DEFENDANT_USER_AUTH_TOKEN)).thenReturn(Optional.of(organisation));
+        when(organisation.getOrganisationIdentifier()).thenReturn(ORG_ID);
+    }
+
+    @Nested
+    class AssignCase {
+
+        @Test
+        void shouldAssignCaseToUser_WhenCalled() {
+            service.assignCase(CASE_ID, DEFENDANT_USER_AUTH_TOKEN);
+
+            verify(organisationService).findOrganisation(DEFENDANT_USER_AUTH_TOKEN);
+
+            CaseAssignedUserRoleWithOrganisation caseAssignedUserRoleWithOrganisation
+                = CaseAssignedUserRoleWithOrganisation.builder()
+                .caseDataId(CASE_ID)
+                .userId(USER_ID)
+                .caseRole(CaseRole.RESPONDENTSOLICITORONE.getFormattedName())
+                .organisationId(ORG_ID)
+                .build();
+
+            AddCaseAssignedUserRolesRequest addCaseAssignedUserRolesRequest = AddCaseAssignedUserRolesRequest.builder()
+                .caseAssignedUserRoles(List.of(caseAssignedUserRoleWithOrganisation))
+                .build();
+
+            verify(caseAccessDataStoreApi).addCaseUserRoles(
+                eq(USER_AUTH_TOKEN),
```

Comment:
no need of "eq(...)" here and other in the same verify call, pass the values directly.